### ISSUE TITLE
Plotting functionality not working

### ISSUE THREAD

**Issue by petebachant** [petebachant/NTNU-HAWT-turbinesFoam#18]
From https://github.com/turbinesFoam/turbinesFoam/issues/297 cc @chandukec

- [ ] `wake` -- needs sampling
- [ ] `spanwise` -- file names not formed correctly
- [ ] Remove `strut-perf` -- only relevant to RM2
- [ ] `recovery` -- might want to remove that since it may not be relevant here

Probably happened when switching over to a single run script, and focusing on sampling nacelle values. Basically we need to do similar plotting at https://github.com/petebachant/UNH-RVAT-turbinesFoam.

**Comment by chandukec** [comment]
In this I hope we are trying to plot both the turbine characteristics? And particularly interested in the wakes, velocity profiles etc. at different axial locations after/beyond the second turbine? What I understood from the Python codes of plot.py, plotting.py, processing.py etc.  is that, instead of sampling at the nacelle we need to sample at different axial location?

**Comment by petebachant** [comment]
Yep. In `run.py` there is a function called `gen_sets_file` which determines where the velocity field is sampled. That function, `sets.template` and the code for loading and plotting the data all needs to be updated. I wanted to leave it templated to allow for programmatically moving the turbines around and having the sampled profiles move with them.

**Comment by chandukec** [comment]
Ok, thanks for the clarification, I will take a closer look. So, are these issues are resolved and the plotting code is ready to use?

**Comment by petebachant** [comment]
Nope, the issues aren't resolved. Sorry I don't have a ton of time to dedicate to supporting this at the moment. If you end up modifying any code to suit your needs in the meantime, a PR would be much appreciated!

**Comment by chandukec** [comment]
Ok, thanks for this information. Although I have no idea about the Python but I will surely give it a try to make my case working. I will update you if this works for me.

**Comment by chandukec** [comment]
Could you kindly suggest some tutorial/case in which these functionalities are working. I need to compare the velocity profile in the wakes of the 2nd turbine at x/D=1,2.5, 4, urgently. For this, I tried postProcess -func sampleDict -latestTime, but unfortunately it haven't helped.

**Comment by chandukec** [comment]
Could you kindly suggest some tutorial/case in which these functionalities are working. I need to compare the velocity profile in the wakes of the 2nd turbine at x/D=1,2.5, 4, urgently. For this, I tried postProcess -func sampleDict -latestTime, but unfortunately it haven't helped.

**Comment by petebachant** [comment]
Something similar is done in [this case](https://github.com/petebachant/UNH-RVAT-turbinesFoam), but instead of using a template file, it generates the whole file from scratch. A hybrid of these two approaches could be used on this case to add new items to the `sets.template` file based on the position of each turbine, and the arrays of points you'd like to sample. Here is the function to create the sets file: https://github.com/petebachant/UNH-RVAT-turbinesFoam/blob/master/run.py#L85

**Comment by chandukec** [comment]
Many thanks for this help. I tried but unfortunately I could not make it, since I am not very familiar with the Python. Instead I took a very naive method to plot the Umag/U0 at the distance of x/R=2 from turbine 2, similar to the plot 6 (a) of the blind test 2 paper. I took a slice on paraview at the axial distance of x/R=2 from turbine 2 and plotted about the line in y axis at z=0. The corresponding figure is attached herewith. Now from the figure it is clearly evident that at y/R=0, there are many fluctuations, so I am not sure if instead of taking z=0 I had to average in the whole z plane and then plot this in y axis. Or in other words am I wrong to plot the variation of non dimensional velocity with the y/R at z=0, instead I should have averaged Umag/U0 in z axis and then have plotted U/U0 in the y/R? If I had to take average in z, then is there any other and simple way to do that in postProcess or in paraview?
![velocity_profile](https://user-images.githubusercontent.com/22544253/75014319-aa736580-548e-11ea-8faf-c47d0e49c39b.png)

**Comment by petebachant** [comment]
I don't believe that the experiments average in the vertical direction, so that is likely not necessary. The fluctuations you see about y/R=0 are likely due to the hub wake.
